### PR TITLE
 Replacing the IsDstDstSrc and IsDstSrcSrc tables with a flag.

### PR DESCRIPTION
**Pull request by tannergooding**
This replaces the two large `IsDstDstSrcAVXInstruction` and `IsDstSrcSrcAVXInstruction` switch statements with flags.

Previous checks had shown these methods taking ~10% of the emit time. Working on getting current numbers to share.

### CODE REVIEW

**Comment by tannergooding** [comment]
FYI. @CarolEidt, @fiigii

File: src/jit/emitxarch.cpp [hunk 1]
```diff
@@ -137,251 +137,8 @@ bool emitter::IsAVXInstruction(instruction ins)
...
-        case INS_vpsravd:
-        case INS_vpsllvd:
-        case INS_vpsllvq:
-        case INS_xorpd:
-        case INS_xorps:
-            return IsAVXInstruction(ins);
-        default:
-            return false;
-    }
+    return ((CodeGenInterface::instInfo[ins] & INS_Flags_IsDstDstSrcAVXInstruction) != 0)
```
Comment (tannergooding) [inline src/jit/emitxarch.cpp]: note to self: remove the `TODO-XArch-Cleanup` comment on this method

**Comment by tannergooding** [comment]
There are some other switch tables that might be good to covert as well, but I figure those would be good for separate investigation.

**Comment by CarolEidt** [comment]
The changes look good overall; I'm currently doing a check of the flag values against the old methods. It would be great if @fiigii could also validate.

These have caused a number of bugs in the past. I'd like to see a combination of jitStress and jitStressRegs={2|3|8} and x86/x64Windows/X64Unix, so that we get these instructions encoded with registers other than the default ymm0.

**Comment by tannergooding** [comment]
@dotnet-bot test Ubuntu x64 Checked jitstress1
@dotnet-bot test Ubuntu x64 Checked jitstressregs0x1000
@dotnet-bot test Ubuntu x64 Checked jitstressregs0x10
@dotnet-bot test Ubuntu x64 Checked jitstressregs0x80
@dotnet-bot test Ubuntu x64 Checked jitstressregs1
@dotnet-bot test Ubuntu x64 Checked jitstressregs2
@dotnet-bot test Ubuntu x64 Checked jitstressregs3
@dotnet-bot test Ubuntu x64 Checked jitstressregs4
@dotnet-bot test Ubuntu x64 Checked jitstressregs8

@dotnet-bot test Ubuntu x64 Checked jitstress2
@dotnet-bot test Ubuntu x64 Checked jitstress2_jitstressregs0x1000
@dotnet-bot test Ubuntu x64 Checked jitstress2_jitstressregs0x10
@dotnet-bot test Ubuntu x64 Checked jitstress2_jitstressregs0x80
@dotnet-bot test Ubuntu x64 Checked jitstress2_jitstressregs1
@dotnet-bot test Ubuntu x64 Checked jitstress2_jitstressregs2
@dotnet-bot test Ubuntu x64 Checked jitstress2_jitstressregs3
@dotnet-bot test Ubuntu x64 Checked jitstress2_jitstressregs4
@dotnet-bot test Ubuntu x64 Checked jitstress2_jitstressregs8

@dotnet-bot test Windows_NT x64 Checked jitstress1
@dotnet-bot test Windows_NT x64 Checked jitstressregs0x1000
@dotnet-bot test Windows_NT x64 Checked jitstressregs0x10
@dotnet-bot test Windows_NT x64 Checked jitstressregs0x80
@dotnet-bot test Windows_NT x64 Checked jitstressregs1
@dotnet-bot test Windows_NT x64 Checked jitstressregs2
@dotnet-bot test Windows_NT x64 Checked jitstressregs3
@dotnet-bot test Windows_NT x64 Checked jitstressregs4
@dotnet-bot test Windows_NT x64 Checked jitstressregs8

@dotnet-bot test Windows_NT x64 Checked jitstress2
@dotnet-bot test Windows_NT x64 Checked jitstress2_jitstressregs0x1000
@dotnet-bot test Windows_NT x64 Checked jitstress2_jitstressregs0x10
@dotnet-bot test Windows_NT x64 Checked jitstress2_jitstressregs0x80
@dotnet-bot test Windows_NT x64 Checked jitstress2_jitstressregs1
@dotnet-bot test Windows_NT x64 Checked jitstress2_jitstressregs2
@dotnet-bot test Windows_NT x64 Checked jitstress2_jitstressregs3
@dotnet-bot test Windows_NT x64 Checked jitstress2_jitstressregs4
@dotnet-bot test Windows_NT x64 Checked jitstress2_jitstressregs8

@dotnet-bot test Windows_NT x86 Checked jitstress1
@dotnet-bot test Windows_NT x86 Checked jitstressregs0x1000
@dotnet-bot test Windows_NT x86 Checked jitstressregs0x10
@dotnet-bot test Windows_NT x86 Checked jitstressregs0x80
@dotnet-bot test Windows_NT x86 Checked jitstressregs1
@dotnet-bot test Windows_NT x86 Checked jitstressregs2
@dotnet-bot test Windows_NT x86 Checked jitstressregs3
@dotnet-bot test Windows_NT x86 Checked jitstressregs4
@dotnet-bot test Windows_NT x86 Checked jitstressregs8

@dotnet-bot test Windows_NT x86 Checked jitstress2
@dotnet-bot test Windows_NT x86 Checked jitstress2_jitstressregs0x1000
@dotnet-bot test Windows_NT x86 Checked jitstress2_jitstressregs0x10
@dotnet-bot test Windows_NT x86 Checked jitstress2_jitstressregs0x80
@dotnet-bot test Windows_NT x86 Checked jitstress2_jitstressregs1
@dotnet-bot test Windows_NT x86 Checked jitstress2_jitstressregs2
@dotnet-bot test Windows_NT x86 Checked jitstress2_jitstressregs3
@dotnet-bot test Windows_NT x86 Checked jitstress2_jitstressregs4
@dotnet-bot test Windows_NT x86 Checked jitstress2_jitstressregs8

**Comment by fiigii** [comment]
This PR removes 251 instructions from the two tables, but adds the flags for 254 instructions in inslist.
Did you find 3 more instructions that should be handled by the VEX.vvvv encoding?

**Comment by tannergooding** [comment]
No, the mappings should be 1-to-1.

> but adds the flags for 254 instructions in inslist.

I see 239 hits for `INS_Flags_IsDstDstSrcAVXInstruction` and and 12 hits for `INS_Flags_IsDstSrcSrcAVXInstruction`, which equals 251.

**Comment by tannergooding** [comment]
The `Windows_NET x64...` failures was due to `baseservices_threading._mutex_misc_waitone2_waitone2_._mutex_misc_waitone2_waitone2_cmd`. Not going to rerun.

**Comment by tannergooding** [comment]
The `Windows_NT x86...` failure is due to an assert, and is unrelated: `Assert failure(PID 5300 [0x000014b4], Thread: 1680 [0x0690]): Assertion failed 'op1->OperGet() == GT_LONG || op1->OperGet() == GT_MUL_LONG' in 'System.Text.OSEncoding:GetMaxByteCount(int):int:this' (IL size 55)` -- CC. @CarolEidt

**Comment by CarolEidt** [comment]
> Looks to be a known issue: #19397

Yes, I've got a PR up to fix it: https://github.com/dotnet/coreclr/pull/19695

**Pull request merged**